IoTHubManager/CreateDeviceTests.cs

```diff
@@ -0,0 +1,262 @@
+         * has system-generated id and correct auth credentails.
+         */
+        [Fact, Trait(Constants.TEST, Constants.INTEGRATION_TEST)]
+        public void DeviceCreated_IfEmptyIdandCustomAuthPassed()
+        {
+            //Arrange
+            string primaryKey = Guid.NewGuid().ToString("N"),
+            string secondaryKey = Guid.NewGuid().ToString("N");
+            // DeviceId must be empty to be auto generated.
+            string device = DEVICE_TEMPLATE_SYMMETRIC_AUTH.Replace(Constants.Keys.DEVICE_ID, "")
+                                                          .Replace(Constants.Keys.PRIMARY_KEY, primaryKey)
+                                                          .Replace(Constants.Keys.SECONDARY_KEY, secondaryKey);
+
+
+            //Act
+            var response = Request.Post(device);
+
+            // Asserts
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            var createdDevice = JObject.Parse(response.Content);
+            var authentication = createdDevice["Authentication"];
+            string createdDeviceId = createdDevice["Id"].ToString();
+            string createdPrimaryKey = authentication["PrimaryKey"].ToString();
+            string createdSecondaryKey = authentication["SecondaryKey"].ToString();
+
+            // Assert device ID and auth or not null OR empty and
+            // other required properties are set. Also check auth type.
+            Assert.False(string.IsNullOrEmpty(createdDeviceId));
+            Assert.Equal(Constants.Auth.SYMMETRIC, authentication["AuthenticationType"]);
+            Assert.Equal(primaryKey, createdPrimaryKey);
+            Assert.Equal(secondaryKey, createdSecondaryKey);
+            Assert.False(createdDevice["IsSimulated"].ToObject<bool>());
+            Assert.True(createdDevice["Enabled"].ToObject<bool>());
+        }
+
+        /**
+         * Creates a device with custom id and X509 auth credentails.
+         * Verifies that the resultant device has correct id
+         * and auth credentails.
+         */
+        [Fact, Trait(Constants.TEST, Constants.INTEGRATION_TEST)]
+        public void DeviceCreated_IfCustomIdandX509AuthPassed()
+        {
+            // Arrange
+            string id = Guid.NewGuid().ToString();
+            string primaryThumbprint = Helpers.GenerateNewThumbPrint();
+            string secondaryThumbprint = Helpers.GenerateNewThumbPrint();
+
+            string device = DEVICE_TEMPLATE_X509_AUTH.Replace(Constants.Keys.DEVICE_ID, id)
+                                                     .Replace(Constants.Keys.PRIMARY_TH, primaryThumbprint)
+                                                     .Replace(Constants.Keys.SECONDARY_TH, secondaryThumbprint);
+
+            // Act
+            var response = Request.Post(device);
+
+            // Asserts 
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            var createdDevice = JObject.Parse(response.Content);
```

Comment:
can you extract some of these checks into a private method? It looks like they are pretty similar

Reply:
Yes, I did those. Now they are Helpers.Device